test/extended/openstack/observability.go

```diff
@@ -0,0 +1,797 @@
+		  )
+		)
+	  )`
+)
+
+var _ = g.Describe("[sig-installer][Suite:openshift/openstack] Check shift-on-stack", func() {
+
+	defer g.GinkgoRecover()
+
+	shiftstackKubeConfig := os.Getenv(shiftstackKubeConfigEnvVarName)
+	rhosoKubeConfig := os.Getenv(rhosoKubeConfigEnvVarName)
+	shiftstackPassFile := os.Getenv(shiftstackPassFileEnvVarName)
+	metricStorageURL := fmt.Sprintf("https://%s/api/v1/query", metricStoragePrometheusSvc)
+	var computeClient *gophercloud.ServiceClient
+	var volumeClient *gophercloud.ServiceClient
+	var shiftstackFederateRouteHost string
+	var shiftstackToken string
+	var err error
+
+	oc := exutil.NewCLI(openstackNamespaceName)
+
+	g.BeforeEach(func(ctx g.SpecContext) {
+		g.By(fmt.Sprintf("Checking the %s, %s and %s env vars are being set", shiftstackKubeConfigEnvVarName, shiftstackPassFileEnvVarName, rhosoKubeConfigEnvVarName))
+		if shiftstackKubeConfig == "" || shiftstackPassFile == "" || rhosoKubeConfig == "" {
+			e2eskipper.Skipf("%s, %s and %s env vars must be set for this test to run", shiftstackKubeConfigEnvVarName, shiftstackPassFileEnvVarName, rhosoKubeConfigEnvVarName)
+		}
+
+		g.By("Getting the Openstack clients")
+		computeClient, err = client.GetServiceClient(ctx, openstack.NewComputeV2)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Failed to get the OpenStack compute client")
+		volumeClient, err = client.GetServiceClient(ctx, openstack.NewBlockStorageV3)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Failed to get the OpenStack block storage client")
+
+		// Get the shiftstack prometheus federate endpoint (in order to scrape metrics from it)
+		g.By(fmt.Sprintf("Getting the '%s' route host from the shiftstack cluster", shiftstackPrometheusFederateRouteName))
+		shiftstackConfig, err := clientcmd.BuildConfigFromFlags("", shiftstackKubeConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		shiftstackRouteClient, err := routev1.NewForConfig(shiftstackConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		shiftstackFederateRoute, err := shiftstackRouteClient.Routes(monitoringNamespaceName).Get(ctx, shiftstackPrometheusFederateRouteName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		shiftstackFederateRouteHost = shiftstackFederateRoute.Status.Ingress[0].Host
+		o.Expect(shiftstackFederateRouteHost).NotTo(o.BeEmpty(), "Empty '%s' route host found", shiftstackPrometheusFederateRouteName)
+		e2e.Logf("Shiftstack federate route host: '%v'", shiftstackFederateRouteHost)
+
+		// Create a token in the shiftstack cluster
+		g.By("Creating a token in the shiftstack cluster")
+		SetTestContextHostFromKubeconfig(shiftstackKubeConfig)
+		pass, err := os.ReadFile(shiftstackPassFile)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		_, err = oc.Run("login").Args("-u", "kubeadmin").InputString(string(pass) + "\n").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		shiftstackToken, err = oc.Run("whoami").Args("-t").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		e2e.Logf("Token: '%s'", shiftstackToken)
```

Comment:
We should probably avoid logging the token.

Reply:
Sure, I've removed it